### DISCUSSION TITLE
Expressions in pre-compiled bins

### DISCUSSION THREAD

**Discussion by pmknowles** [arendst/Tasmota#19674] [Q&A]
A user has come on the old Sonoff group with an issue with Expressions. He has a pre-compiled ESP32 bin which seems to be working with Expressions. Is it now part of the pre-compiled bins? The features and builds page implies it shouldn't be.

**Comment by sfromis** [comment]
Yea, my experience is that `USE_EXPRESSION` and `SUPPORT_IF_STATEMENT` are included in ESP32 standard config.

Looks like builds not for ESP8266_1M are [always enabling](https://github.com/arendst/Tasmota/blob/development/tasmota/include/tasmota_configurations.h#L1045-L1060) those two and also `SUPPORT_MQTT_EVENT` plus options for file system, SPI, and ping.

**Comment by pmknowles** [comment]
Can someone update the documents please? I don’t want to do it myself when I have no personal evidence of it.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows

**Reply by barbudor** [reply]
Done
![image](https://github.com/arendst/Tasmota/assets/2996042/ba366331-a33d-4455-ad67-169d697a6de3)

And same for IF

**Comment by pmknowles** [comment]
Thank you

Regards,

Phil
